pkg/controller/remotemachineset/remotemachineset_controller.go

```diff
@@ -575,13 +571,19 @@ func (c *Controller) deleteFinalizer(clusterDeployment *cov1.ClusterDeployment)
 	return err
 }
 
-func (c *Controller) addFinalizer(clusterDeployment *cov1.ClusterDeployment) error {
+func (c *Controller) setMachineSetsSyncedStatus(clusterDeployment *cov1.ClusterDeployment) error {
 	clusterDeployment = clusterDeployment.DeepCopy()
 	controller.AddFinalizer(clusterDeployment, cov1.FinalizerRemoteMachineSets)
+	clusterDeployment.Status.RemoteMachineSetsSynced = true
+	clusterDeployment.Status.RemoteMachineSetsSyncedGeneration = clusterDeployment.Generation
```

Comment:
I think that may leave these new fields as unused, do you think they still provide some value in having set? Otherwise we can just assume this is the state if the finalizer has been added.